components/calendar/CalWeeklyDay.vue

```diff
@@ -0,0 +1,323 @@
+     */
+    displayYear: Number,
+    /**
+     * The displaed month.
+     */
+    displayMonth: Number,
+    /**
+     * Today's date, used to determine if this day should be highlighted yellow.
+     */
+    todaysDate: String,
+    /**
+     * The current calendar view, "monthly", "weekly", or "upcoming".
+     */
+    view: String
+  },
+  data() {
+    return {
+      /**
+       * Hovering over event.
+       */
+      detailedOpen: false,
+      /**
+       * Hovering over detail box.
+       */
+      rolloverDetailed: false,
+      /**
+       * Info to pass to detailed view.
+       */
+      detailedInfo: {},
+      /**
+       * Maximum number of concurrent events to show.
+       */
+      maxConcurrent: 0,
+      /**
+       * Total number of half hours slots to show in each day (in weekly view).
+       */
+      totalHalfHours: 48
+    };
+  },
+  computed: {
+    /**
+     * Gets the events for this day.
+     *
+     * @returns {Array} An array of the event objects for this day.
+     */
+    events() {
+      // get events on this day
+      const stringDate = this.getStringDate();
+      const newEvents = [];
+
+      for (let i = 0; i < this.info.length; i++) {
+        if (this.info[i].date_utc.split(' ')[0] === stringDate) {
+          newEvents.push(this.info[i]);
+        }
+      }
+      return newEvents;
+    }
+  },
+  methods: {
+    /**
+     * Generates the string representation of a date.
+     *
+     * @returns {string} The date as a string.
+     */
+    getStringDate() {
+      return getStringDate(this.displayMonth, this.date, this.displayYear);
+    },
+    /**
+     * Set detailedOpen to false after 0.1 seconds, to allow time for potential mouseover.
+     */
+    startDetailClose() {
+      setTimeout(() => {
+        if (!this.rolloverDetailed) {
+          this.detailedOpen = false;
+        }
+      }, 100);
+    },
+    /**
+     * Removes the duplicate events by title.
+     *
+     * @param {Array} events An array of the events on this day.
+     * @returns {Array} The events with duplicates removed.
+     */
+    removeDuplicates(events) {
+      const eventsDict = {};
+      for (let i = 0; i < events.length; i++) {
+        const title = events[i].title;
+        eventsDict[title] = events[i];
+      }
+      const retEvents = Object.keys(eventsDict).map((k) => {
+        return eventsDict[k];
+      });
+      return retEvents;
+    },
+    /**
+     * Organizes events into their respective timeslots and handles overlapping events.
+     *
+     * @param {Array} events An array of the events on this day.
+     */
+    getEventsDict(events) {
+      const eventsDict = {}; // { time/tr index : [ event ] }; (ea. event only appears in FIRST row / start time)
+      const calcConcurrentDict = {}; // separate copy of eventsDict (ea. event appears in ALL rows in timespan)
+
+      const eventToOverlapDict = {}; // { eventId: numOverlap } (holds # events overlapping given event)
```

Comment:
this method is also in `CalMonthDay` (I haven't checked if it's exactly the same, but it looks similar), abstract it/move to utils

Reply:
it seems the methods were copied and pasted between the two files. the two files each use a unique subset of the methods. i deleted the unused methods from each file.